### PR TITLE
Add option to skip generator caching

### PR DESCRIPTION
**Pull request by mj-will**
Following some discussion with @spxiwh, I wanted to add an option to disable the generator caching. For now, I've taken the simple approach of having two functions but I'm open to suggestions for how to best do this.

I've also added a test to ensure the cache is still called correctly.

### CODE REVIEW

File: BBHX_Phenom.py [hunk 1]
```diff
@@ -269,9 +277,18 @@ def _bbhx_fd(
     # To solve this we *round* the *logarithm* of this mass-dependent start
     # frequency. The factor of 25 ensures reasonable spacing while doing this.
     # So we round down to the nearest 1/25 of the logarithm of the frequency
-    log_mf_min = int(math.log(f_min*MTSUN_SI*(m1+m2)) * 25)
-
-    wave_gen = get_waveform_genner(log_mf_min, run_phenomd=run_phenomd)
+    log_mf_min = math.log(f_min*MTSUN_SI*(m1+m2)) * 25
```
Comment (WuShichao) [inline BBHX_Phenom.py:280]: Thanks for the PR! Why do you remove `int()`? Previously, we used it to round down.

File: BBHX_Phenom.py [hunk 2]
```diff
@@ -269,9 +277,18 @@ def _bbhx_fd(
...
+    if cache_generator:
+        # Use int to round down
+        wave_gen = cached_get_waveform_genner(
+            int(log_mf_min),
```
Comment (WuShichao) [inline BBHX_Phenom.py:284]: Ah, OK, you moved it here.

**Review by WuShichao — APPROVED** [review]
Looks good!

File: BBHX_Phenom.py [hunk 3]
```diff
@@ -129,6 +134,7 @@ def _bbhx_fd(
     direct=False,
     num_interp=100,
     interp_f_lower=1e-4,
+    cache_generator=True,
```
Comment (spxiwh) [inline BBHX_Phenom.py:137]: I would suggest to make the default `False`. Easier to avoid issues if the user has to "opt-in" to use this

**Pull request merged**